test/smoke/testApps/SpringBootAuto/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/SpringSchedulerTest.java

```diff
@@ -0,0 +1,84 @@
+package com.microsoft.applicationinsights.smoketest;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import com.google.common.base.Stopwatch;
+import com.microsoft.applicationinsights.internal.schemav2.Data;
+import com.microsoft.applicationinsights.internal.schemav2.Envelope;
+import com.microsoft.applicationinsights.internal.schemav2.RemoteDependencyData;
+import com.microsoft.applicationinsights.internal.schemav2.RequestData;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+@UseAgent
+public class SpringSchedulerTest extends AiSmokeTest {
+
+    @Test
+    @TargetUri("/scheduler")
+    public void fixedRateSchedulerTest() throws Exception {
+        Stopwatch stopwatch = Stopwatch.createStarted();
+        while (mockedIngestion.getCountForType("RequestData") < 20 && stopwatch.elapsed(TimeUnit.SECONDS) < 10) {
+        }
+
+        List<Envelope> rdList = mockedIngestion.getItemsEnvelopeDataType("RequestData");
+        assertTrue(rdList.size() >= 20);
+
+        List<Envelope> httpRequestList = new ArrayList<>(1);
+        List<Envelope> schedulerRequestList = new ArrayList<>();
+        groupRequestList(httpRequestList, schedulerRequestList, rdList);
+        assertEquals(1, httpRequestList.size());
+        assertTrue(schedulerRequestList.size() >= 20);
```

Comment:
should this be 19, since total is 20?
```suggestion
        assertTrue(schedulerRequestList.size() >= 19);
```

Reply:
it's always greater than 30.. because fixedRate = 100 miliseconds.  It's never less than 20

Reply:
every run it's around 35

Reply:
only waiting for 20 above:

`mockedIngestion.getCountForType("RequestData") < 20`